File: MkdirChmodRmdir/MkdirChmodRmdir.rb
require 'fileutils'

class MkdirChmodRmdir
  def initialize(opt = {})
      @dir = "ruby_dir"
  end

  def run
    if not Dir.exist?(@dir)
      puts("mkdir")
      FileUtils.mkdir_p(@dir)
    end

    if Dir.exist?(@dir)
      puts("chmod")
      FileUtils.chmod(0o777, @dir)

      puts("rmdir")
      FileUtils.rmdir(@dir)
    end
  end
end

if $0 == __FILE__
  mcr = MkdirChmodRmdir.new()
  mcr.run()
end

File: MkdirChmodRmdir/MkdirChmodRmdir.py
import os
class MkdirChmodRmdir:
  def __init__(self):
    self.dir = "python_dir"

  def run(self):
    if(not os.path.exists(self.dir)):
      os.mkdir(self.dir)
      print("mkdir")

    if(os.path.exists(self.dir)):
      print("chmod")
      os.chmod(self.dir, 0o777)

      print("rmdir")
      os.rmdir(self.dir)

if __name__ == '__main__':
  mcr = MkdirChmodRmdir()
  mcr.run()
